receiver/splunkhecreceiver/receiver.go

```diff
@@ -0,0 +1,292 @@
+	errGzipReaderRespBody    = initJSONResponse(responseErrGzipReader)
+	errUnmarshalBodyRespBody = initJSONResponse(responseErrUnmarshalBody)
+	errNextConsumerRespBody  = initJSONResponse(responseErrNextConsumer)
+)
+
+// splunkReceiver implements the component.MetricsReceiver for Splunk HEC metric protocol.
+type splunkReceiver struct {
+	sync.Mutex
+	logger       *zap.Logger
+	config       *Config
+	nextConsumer consumer.MetricsConsumer
+	server       *http.Server
+}
+
+var _ component.MetricsReceiver = (*splunkReceiver)(nil)
+
+// New creates the Splunk HEC receiver with the given configuration.
+func New(
+	logger *zap.Logger,
+	config Config,
+	nextConsumer consumer.MetricsConsumer,
+) (component.MetricsReceiver, error) {
+	if nextConsumer == nil {
+		return nil, errNilNextConsumer
+	}
+
+	if config.Endpoint == "" {
+		return nil, errEmptyEndpoint
+	}
+
+	r := &splunkReceiver{
+		logger:       logger,
+		config:       &config,
+		nextConsumer: nextConsumer,
+		server: &http.Server{
+			Addr: config.Endpoint,
+			// TODO: Evaluate what properties should be configurable, for now
+			//		set some hard-coded values.
+			ReadHeaderTimeout: defaultServerTimeout,
+			WriteTimeout:      defaultServerTimeout,
+		},
+	}
+
+	return r, nil
+}
+
+// StartMetricsReception tells the receiver to start its processing.
+// By convention the consumer of the received data is set when the receiver
+// instance is created.
+func (r *splunkReceiver) Start(_ context.Context, host component.Host) error {
+	r.Lock()
+	defer r.Unlock()
+
+	var ln net.Listener
+	// set up the listener
+	ln, err := r.config.HTTPServerSettings.ToListener()
+	if err != nil {
+		return fmt.Errorf("failed to bind to address %s: %w", r.config.Endpoint, err)
+	}
+
+	mx := mux.NewRouter()
+	mx.HandleFunc(hecPath, r.handleReq)
+
+	r.server = r.config.HTTPServerSettings.ToServer(mx)
+
+	// TODO: Evaluate what properties should be configurable, for now
+	//		set some hard-coded values.
+	r.server.ReadHeaderTimeout = defaultServerTimeout
+	r.server.WriteTimeout = defaultServerTimeout
+
+	go func() {
+		if errHTTP := r.server.Serve(ln); errHTTP != nil {
+			host.ReportFatalError(errHTTP)
+		}
+	}()
+
+	return err
+}
+
+// StopMetricsReception tells the receiver that should stop reception,
+// giving it a chance to perform any necessary clean-up.
+func (r *splunkReceiver) Shutdown(context.Context) error {
+	r.Lock()
+	defer r.Unlock()
+
+	err := r.server.Close()
+
+	return err
+}
+
+func (r *splunkReceiver) handleReq(resp http.ResponseWriter, req *http.Request) {
+	transport := "http"
+	if r.config.TLSSetting != nil {
+		transport = "https"
+	}
+	ctx := obsreport.ReceiverContext(req.Context(), r.config.Name(), transport, r.config.Name())
+	ctx = obsreport.StartMetricsReceiveOp(ctx, r.config.Name(), transport)
+
+	if req.Method != http.MethodPost {
+		r.failRequest(ctx, resp, http.StatusBadRequest, invalidMethodRespBody, nil)
+		return
```

Comment:
I'm not sure how the obsreport.StartMetricsReceiveOp thing works, but is it necessary to signal the end of the operation or is that handled automatically by the `reqSpan.End()` thing in `failRequest`?